### PR TITLE
STM32G43XX Analog Comparator

### PR DESCRIPTION
**Pull request by daniel-p-carvalho**
## Summary

Add STM32G43XX Analog Comparator driver.
Add nucleo-g431kb Analog Comparator example.

## Testing

Tested using nucleo-g431kb.

### CODE REVIEW

File: arch/arm/src/stm32/stm32_comp.c [hunk 1]
```diff
@@ -34,6 +46,8 @@
```
Comment (hartmannathan) [inline arch/arm/src/stm32/stm32_comp.c:46]: The above comment is now incorrect.

I recommend:

```
/* This file is only a thin shell that includes the correct COMP
 * implementation. At this moment STM32 COMP IP versions 1 and 2 are
 * supported.
 *   - STM32 COMP IP version 1: SMT32F33XX
 *   - STM32 COMP IP version 2: SMT32G4XXX
 */
```

File: arch/arm/src/stm32/stm32_comp.h [hunk 2]
```diff
@@ -192,5 +84,4 @@ FAR struct comp_dev_s *stm32_compinitialize(int intf);
 #endif
 #endif /* __ASSEMBLY__ */
 
-#endif /* CONFIG_STM23_COMP */
-#endif /* __ARCH_ARM_SRC_STM32_STM32_COMP_H */
+#endif /* __ARCH_ARM_SRC_STM32_STM32_COMP_H */
```
Comment (hartmannathan) [inline arch/arm/src/stm32/stm32_comp.h:87]: Recommend adding newline at end of file.

**Review by hartmannathan — COMMENTED** [review]
Thank you for doing this

File: arch/arm/src/stm32/stm32_comp.c (hunk 1 above)
Comment (hartmannathan) [reply arch/arm/src/stm32/stm32_comp.c:46]: It appears GitHub isn't showing the comment. It's on line 41 (was line 29) of arch/arm/src/stm32/stm32_comp.c and currently reads:

```
/* This file is only a thin shell that includes the correct COMP
 * implementation. At this moment only STM32 COMP IP version 1 device is
 * suportted.
 *   - STM32 COMP IP version 1: SMT32F33XX
 */
```

File: arch/arm/src/stm32/stm32_comp.h (hunk 2 above)
Comment (acassis) [reply arch/arm/src/stm32/stm32_comp.h:87]: > Recommend adding newline at end of file.

I think it is not necessary, just look others header .h files in the same directory and at include/

File: arch/arm/src/stm32/stm32_comp.c (hunk 1 above)
Comment (daniel-p-carvalho) [reply arch/arm/src/stm32/stm32_comp.c:47]: Thanks for helping the review. Done

File: arch/arm/src/stm32/stm32_comp.h (hunk 2 above)
Comment (daniel-p-carvalho) [reply arch/arm/src/stm32/stm32_comp.h]: Thanks for helping the review. Done

Comment (daniel-p-carvalho) [reply arch/arm/src/stm32/stm32_comp.h]: Should I revert the commit 60c5ec0?

Comment (acassis) [reply arch/arm/src/stm32/stm32_comp.h]: Don't need to

**Pull request merged**